conda_smithy/lint_recipe.py

```diff
@@ -756,8 +756,12 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
 
         url = None
         for source_section in sources_section:
-            if "url" in source_section and source_section["url"].startswith(
-                "https://pypi.io/packages/source/"
+            if (
+                "url" in source_section
+                and isinstance(source_section["url"], str)
+                and source_section["url"].startswith(
+                    "https://pypi.io/packages/source/"
+                )
             ):
```

Comment:
It's up to you but I'd be tempted to simplify this to:
```suggestion
            if str(source_section.get("url")).startswith("https://pypi.io/packages/source/"):
```